java-client/session/src/main/java/io/deephaven/client/impl/SharedId.java

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.client.impl;
+
+import com.google.protobuf.ByteString;
+import io.deephaven.proto.util.ByteHelper;
+import io.deephaven.proto.util.SharedTicketHelper;
+
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * An opaque holder for a shared object ID.
+ */
+public class SharedId implements HasTicketId, HasPathId {
+
+    /**
+     * @return a new secure random {@link SharedId} based on {@link UUID#randomUUID()}.
```

Comment:
I don't think mentioning `UUID` is necessarily wanted / clear. It _may_ give off the appearance that we are generating a UTF-8 / toString version of UUID.

I might just just use the wording the `UUID#newRandom` doc does:

> cryptographically strong pseudo random number generator